asmcomp/asmlink.ml

```diff
@@ -49,40 +49,15 @@ let cmx_required = ref ([] : CU.t list)
 
 let check_consistency file_name unit crc =
   begin try
```

Comment:
There is a check upstream that is missing here, is it intentional? I think it's needed for L60 change below, see https://github.com/ocaml/ocaml/pull/11635.

```suggestion
  begin try
    let source = List.assoc unit.ui_name !implementations_defined in
    raise (Error(Multiple_definition(unit.ui_name, file_name, source)))
  with Not_found -> ()
  end;
  begin try
```

Reply:
Looks like the check is still there, only moved to line 90: https://github.com/ocaml-flambda/ocaml-jst/blob/60d0234646097c947f5bed057a9f9d51a2eaa5c7/asmcomp/asmlink.ml#L90

(Not sure why the check is in a different place. Probably doesn't have to be.)

Reply:
The upstream PR#11635 moves the check from L90 to L60 because otherwise the  now-unguarded crc check will give strange errors.

Reply:
Ah, good catch. I guess we should move it here, then? (Actually, how did the new version of the check end up further down to begin with? Did we backport that PR but move the check?)

Reply:
I've moved the check up, @lukemaurer please confirm